cli/demo: fix \demo node restart

otan opened the pull request:
Also re-enabled the tcl test, as TLS 1.3 has been disabled.

Resolves #49381

Release note (bug fix): Fixed a bug where `\demo node restart` would not
work due to an invalid certificate directory.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49382)
<!-- Reviewable:end -->

knz reviewed (approved):
nice thank you :100:

otan committed:
cli/demo: fix \demo node restart

Also re-enabled the tcl test, as TLS 1.3 has been disabled.

Release note (bug fix): Fixed a bug where `\demo node restart` would not
work due to an invalid certificate directory.

otan commented:
bors r=knz

tftr! examples-orms flake in the red.
